export/export.py

```diff
@@ -0,0 +1,866 @@
+
+        # Process each stage in the pipeline
+        for i in range(start_index, len(self._pipeline)):
+            stage_name = self._pipeline[i]
+            self._current_stage_index = i
+
+            # Get the primary input and configuration parameters for this stage
+            primary_input, config_params = self._get_stage_inputs(stage_name)
+
+            # Run the stage
+            result = self._run_stage(stage_name, primary_input, config_params)
+
+            # Store the result
+            self._store_stage_result(stage_name, result)
+
+    def export(self) -> None:
+        """
+        Execute the full export process.
+
+        This method orchestrates the export process with optional quantization:
+        1. (Optional) Apply quantization to the model
+        2. Export the PyTorch model to ExportedProgram
+        3. Transform and lower to EdgeProgramManager
+        4. Convert to ExecutorchProgramManager
+        """
+        # Start with source transform stage if available
+        if "source_transform" in self._stages:
+            self._run_pipeline("source_transform")
+        # Otherwise start with quantize stage if available
+        elif "quantize" in self._stages:
+            self._run_pipeline("quantize")
+        # Otherwise start with export stage
+        else:
+            self._run_pipeline("export")
+
+    def save_pte_file(self, path: str) -> None:
+        """
+        Save the exported program to a PTE file.
+
+        Args:
+            path: Path where the PTE file will be saved
+
+        Raises:
+            RuntimeError: If the executorch program manager is not initialized
+        """
+        if self._executorch_program_manager is None:
+            raise RuntimeError(
+                "Executorch program manager is not initialized. Run export() first."
+            )
+        self._executorch_program_manager.save(path)
+
+    def get_executorch_program(self) -> Program:
+        """
+        Get the ExecutorchProgram from the ExecutorchProgramManager.
+
+        Returns:
+            The ExecutorchProgram
+
+        Raises:
+            RuntimeError: If the executorch program manager is not initialized
+        """
+        if self._executorch_program_manager is None:
+            raise RuntimeError(
+                "Executorch program manager is not initialized. Run export() first."
+            )
+        return self._executorch_program_manager.executorch_program
+
+    def get_pte_buffer(self) -> bytes:
+        """
+        Get the PTE buffer as bytes.
+
+        Returns:
+            The PTE buffer as bytes
+
+        Raises:
+            RuntimeError: If the executorch program manager is not initialized
+        """
+        if self._executorch_program_manager is None:
+            raise RuntimeError(
+                "Executorch program manager is not initialized. Run export() first."
+            )
+        return self._executorch_program_manager.buffer
+
+    def get_example_input(
```

Comment:
I feel like this and `run_method` seem a bit confusing to have in the ExportSession?

Reply:
The way i'm thinking is that export session lets the users do almost all the things they'd like to with an executorch model like load the model and run via pybindings, get reference inputs etc. and that's why I added these helper utils.